src/planwise/client/current_project/subs.cljs

```diff
@@ -20,6 +25,42 @@
    (reaction (get-in @db [:current-project :project-data]))))
 
 (register-sub
+ :current-project/read-only?
+ (fn [db [_]]
+   (let [current-data (subscribe [:current-project/current-data])]
+     (reaction (:read-only @current-data)))))
+
+(register-sub
+ :current-project/shares
+ (fn [db [_]]
+   (reaction (get-in @db [:current-project :shares]))))
+
+(register-sub
+ :current-project/shares-search-string
+ (fn [db [_]]
+   (reaction (get-in @db [:current-project :sharing :shares-search-string]))))
+
+(register-sub
+ :current-project/filtered-shares
+ (fn [db [_]]
+   (let [search-string (subscribe [:current-project/shares-search-string])
+         shares (subscribe [:current-project/shares])]
+     (reaction
+       (filterv #(gstring/caseInsensitiveContains (:user-email %) (or @search-string "")) (asdf/value @shares))))))
+
+(register-sub
+ :current-project/share-link
+ (fn [db [_]]
+   (reaction
+     (let [token (get-in @db [:current-project :sharing :token])
+           project-id (get-in @db [:current-project :project-data :id])]
+       (asdf/update token (fn [t]
+                            (str
+                              (.-origin js/document.location)
+                              (routes/project-access {:id project-id
+                                                      :token t}))))))))
```

Comment:
Why would you update the token value here? This could lead to an invalid link if the reaction runs twice. Why not simply return the link as the value of the reaction?